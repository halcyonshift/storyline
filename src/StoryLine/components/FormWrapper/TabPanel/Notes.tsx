import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemButton from '@mui/material/ListItemButton'
import ListItemText from '@mui/material/ListItemText'
import Stack from '@mui/material/Stack'
import { useTranslation } from 'react-i18next'
import TooltipIconButton from '@sl/components/TooltipIconButton'
import { GLOBAL_ICONS } from '@sl/constants/icons'
import useTabs from '@sl/layouts/Work/Tabs/useTabs'
import { status } from '@sl/theme/utils'
import { TabPanelProps } from './types'

const NotesPanel = ({ notes }: TabPanelProps) => {
    const { loadTab } = useTabs()
    const { t } = useTranslation()

    return (
        <List disablePadding sx={{ marginLeft: '1px' }}>
            {notes.map((note) => (
                <ListItem
                    key={`note-${note.id}`}
                    divider
                    disablePadding
                    disableGutters
                    sx={{
                        borderLeft: `8px solid ${note.color || status(note.status, 50).color}`,
                        backgroundColor: status(note.status, 50).color
                    }}
                    secondaryAction={
                        <Stack spacing={0} direction='row' className='mr-3'>
                            <TooltipIconButton
                                size='small'
                                text='layout.work.panel.note.edit'
                                link={`/work/${note.work.id}/note/${note.id}/edit`}
                                icon={GLOBAL_ICONS.edit}
                            />
                            <TooltipIconButton
                                size='small'
                                text='layout.work.panel.note.delete'
                                icon={GLOBAL_ICONS.delete}
                                confirm={t('layout.work.panel.note.deleteConfirm', {
                                    name: note.title
                                })}
                                onClick={() => {
                                    note.delete()
                                }}
                            />
                        </Stack>
                    }>
                    <ListItemButton
                        onClick={() => {
                            loadTab({
                                id: note.id,
                                label: note.displayName,
                                link: `note/${note.id}`
                            })
                        }}>
                        <ListItemText
                            primary={note.displayName}
                            secondary={note.date ? note.displayDate : null}
                        />
                    </ListItemButton>
                </ListItem>
            ))}
        </List>
    )
}

export default NotesPanel
