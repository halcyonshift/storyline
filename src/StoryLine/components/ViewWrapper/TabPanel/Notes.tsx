import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemButton from '@mui/material/ListItemButton'
import ListItemText from '@mui/material/ListItemText'
import useTabs from '@sl/layouts/Work/Tabs/useTabs'
import { status, getHex, textColor } from '@sl/theme/utils'
import { NotesPanelProps } from './types'

const NotesPanel = ({ notes }: NotesPanelProps) => {
    const { loadTab } = useTabs()

    return (
        <List disablePadding sx={{ marginLeft: '1px' }}>
            {notes.map((note) => (
                <ListItem
                    key={`note-${note.id}`}
                    divider
                    disablePadding
                    disableGutters
                    sx={{
                        borderLeft: `8px solid ${status(note.status, 500).color}`
                    }}>
                    <ListItemButton
                        sx={
                            note.color
                                ? {
                                      backgroundColor: note.color,
                                      color: textColor(note.color, getHex('white'), getHex('black'))
                                  }
                                : {}
                        }
                        onClick={() => {
                            loadTab({ id: note.id, mode: 'note' })
                        }}>
                        <ListItemText
                            primary={note.displayName}
                            secondary={note.date ? note.displayDate : null}
                        />
                    </ListItemButton>
                </ListItem>
            ))}
        </List>
    )
}

export default NotesPanel
