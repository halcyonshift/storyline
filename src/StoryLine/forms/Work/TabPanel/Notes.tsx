import { IconButton } from '@mui/material'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemButton from '@mui/material/ListItemButton'
import ListItemText from '@mui/material/ListItemText'
import Stack from '@mui/material/Stack'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import { GLOBAL_ICONS } from '@sl/constants/icons'
import { TabPanelProps } from './types'

const NotesPanel = ({ notes }: TabPanelProps) => {
    const { t } = useTranslation()
    const navigate = useNavigate()

    return (
        <List>
            {notes.map((note) => (
                <ListItem
                    key={`image-${note.id}`}
                    divider
                    disablePadding
                    disableGutters
                    secondaryAction={
                        <Stack spacing={2} direction='row'>
                            <IconButton
                                onClick={() => {
                                    navigate(
                                        // eslint-disable-next-line max-len
                                        `/works/${note.work.id}/note/${note.id}/edit`
                                    )
                                }}
                                aria-label={t('layout.work.panel.note.edit')}>
                                {GLOBAL_ICONS.edit}
                            </IconButton>
                            <IconButton edge='end' aria-label={t('layout.work.panel.note.delete')}>
                                {GLOBAL_ICONS.delete}
                            </IconButton>
                        </Stack>
                    }>
                    <ListItemButton
                        onClick={() => navigate(`/works/${note.work.id}/note/${note.id}`)}>
                        <ListItemText
                            primary={note.displayName}
                            secondary={note.date ? note.displayDate : null}
                        />
                    </ListItemButton>
                </ListItem>
            ))}
        </List>
    )
}

export default NotesPanel
