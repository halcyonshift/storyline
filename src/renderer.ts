import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
