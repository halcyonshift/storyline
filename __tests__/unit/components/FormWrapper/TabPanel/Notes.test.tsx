import renderer from 'react-test-renderer'
import Notes from '@sl/components/FormWrapper/TabPanel/Notes'

describe('<TabPanel.Notes />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Notes notes={[]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
