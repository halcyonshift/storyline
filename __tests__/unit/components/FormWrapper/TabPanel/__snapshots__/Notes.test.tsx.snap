// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel.Notes /> renders correctly 1`] = `
<ul
  className="MuiList-root css-zd7zie-MuiList-root"
/>
`;
